tests/phpunit/tests/block-bindings/postMetaSource.php

```diff
@@ -0,0 +1,263 @@
+	 */
+	private function updateGlobalPost() {
+		global $post;
+		$post = self::$post;
+	}
+
+	/**
+	 * Set up for every method.
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$post               = $factory->post->create_and_get();
+		self::$wp_meta_keys_saved = isset( $GLOBALS['wp_meta_keys'] ) ? $GLOBALS['wp_meta_keys'] : array();
+	}
+
+	/**
+	 * Tear down for every method.
+	 */
+	public static function wpTearDownAfterClass() {
+		$GLOBALS['wp_meta_keys'] = self::$wp_meta_keys_saved;
+	}
+
+	/**
+	 * Tests that a block connected to a custom field renders its value.
+	 *
+	 * @ticket 60651
+	 */
+	public function test_custom_field_value_is_rendered() {
+		register_meta(
+			'post',
+			'tests_custom_field',
+			array(
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'string',
+				'default'      => 'Custom field value',
+			)
+		);
+
+		$content = $this->getModifiedPostContent( '<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"core/post-meta","args":{"key":"tests_custom_field"}}}}} --><p>Fallback value</p><!-- /wp:paragraph -->' );
+		$this->assertSame(
+			'<p>Custom field value</p>',
+			$content,
+			'The post content should show the value of the custom field . '
+		);
+	}
+
+	/**
+	 * Tests that a blocks connected in a password protected post don't render the value.
+	 *
+	 * @ticket 60651
+	 */
+	public function test_custom_field_value_is_not_shown_in_password_protected_posts() {
+		register_meta(
+			'post',
+			'tests_custom_field',
+			array(
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'string',
+				'default'      => 'Custom field value',
+			)
+		);
+
+		function wp_tests_require_post_password() {
+			return true;
+		}
+
+		add_filter(
+			'post_password_required',
+			'wp_tests_require_post_password'
+		);
+
+		$content = $this->getModifiedPostContent( '<!-- wp:paragraph {"metadata":{"bindings":{"content":{"source":"core/post-meta","args":{"key":"tests_custom_field"}}}}} --><p>Fallback value</p><!-- /wp:paragraph -->' );
+
+		remove_filter(
+			'post_password_required',
+			'wp_tests_post_password_required'
+		);
+
+		$this->assertSame(
+			'<p>Fallback value</p>',
+			$content,
+			'The post content should show the fallback value instead of the custom field value.'
+		);
+	}
+
+	/**
+	 * Tests that a blocks connected in a post that is not publicly viewable don't render the value.
+	 *
+	 * @ticket 60651
+	 */
+	public function test_custom_field_value_is_not_shown_in_non_viewable_posts() {
+		register_meta(
+			'post',
+			'tests_custom_field',
+			array(
+				'show_in_rest' => true,
+				'single'       => true,
+				'type'         => 'string',
+				'default'      => 'Custom field value',
+			)
+		);
+
+		function wp_tests_make_post_status_not_viewable() {
+			return false;
+		}
+
+		add_filter(
+			'is_post_status_viewable',
+			'wp_tests_make_post_status_not_viewable'
+		);
```

Comment:
```suggestion
		add_filter( 'is_post_status_viewable', '__return_false' );
```

Reply:
Right, these helper functions are handy 👍🏻

Reply:
Done in [this commit](https://github.com/WordPress/wordpress-develop/pull/6197/commits/6e37a4c72e2dc5bd5271835dc340c95958653e78). I couldn't do it through GitHub interface 🤷